components/service-operator/apis/database/v1beta1/postgres_types.go

```diff
@@ -177,23 +187,34 @@ func (p *Postgres) GetStackTemplate() (*cloudformation.Template, error) {
 		instanceCount = DefaultInstanceCount
 	}
 	for i := 0; i < instanceCount; i++ {
-		template.Resources[fmt.Sprintf("%s%d", PostgresResourceInstance, i)] = &cloudformation.AWSRDSDBInstance{
+		instance := cloudformation.AWSRDSDBInstance{
 			DBClusterIdentifier:     cloudformation.Ref(PostgresResourceCluster),
 			DBInstanceClass:         coalesce(p.Spec.AWS.InstanceType, DefaultClass),
 			Engine:                  Engine,
-			EngineVersion:           EngineVersion,
 			PubliclyAccessible:      false,
 			DBParameterGroupName:    cloudformation.Ref(PostgresResourceParameterGroup),
 			Tags:                    tags,
 			DBSubnetGroupName:       cloudformation.Ref(DBSubnetGroupNameParameterName),
 			DeleteAutomatedBackups:  false,
 			CACertificateIdentifier: "rds-ca-2019",
 		}
+
+		// for backwards compatibility only specify EngineVersion if the k8s object specifies it
+		if len(p.Spec.AWS.EngineVersion) > 0 {
```

Comment:
Now that I think about it...this if statement actually isn't necessary. Inlining it will probably do The Right Thing:tm:. Alas, my AWS access has been revoked already so I can't run the tests to confirm it.